scripts/redirectURLParser.js

```diff
@@ -1,17 +1,19 @@
 import i18n from '../locales/i18n.js';
 import pages from '../data/pages.js';
 
-function langURLParser(state) {
+function redirectURLParser(location) {
   var redirectQuery = false;
-  var queryString = state.query;
+  var query = location.query;
 
-  if (queryString) {
-    redirectQuery = queryString.redirect;
+  if (query) {
+    redirectQuery = query.redirect;
   }
 
   // We don't need to redirect.
-  if (!redirectQuery && state.pathname !== "/") {
-    return;
+  if (!redirectQuery && location.pathname !== "/") {
+    return {
+      pathname: ""
+    };
```

Comment:
returning a more predictable value in general from this function, and trying to keep the return type the same. This is also better for testing.